File: Services/DisciplineService/Business/Models/DisciplineModel.cs
using Data_access.Enums;
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace Business.Models
{
    public class DisciplineModel : BaseModel
    {
        public string Name { get; set; }
        public string Description { get; set; }
        public int Course { get; set; }
        public CreditType CreditType { get; set; }
        public int Hours { get; set; }
        public bool IsSelective { get; set; }
    }
}

File: Client/src/app/modules/management/models/discipline.model.ts
import { CreditType } from "../enums/credit-type.model";

export interface IDiscipline {
    id: number;
    name: string;
    description: string;
    course: number;
    creditType: number;
    hours: number;
    isSelective: boolean;
    universityId: number;
    catalogId?: number;
    practicians?: number | undefined;
    lecturers?: number;
}

export interface ISaveDiscipline {
    name: string;
    description: string;
    course: number;
    creditType: number;
    hours: number;
    isSelective: boolean;
    universityId: number;
    catalogId?: number;
}